mis_builder/models/mis_report.py

```diff
@@ -892,18 +904,14 @@ def declare_and_compute_period(self, kpi_matrix,
                                             query argument and returns a
                                             domain compatible with the query
                                             underlying model
+        :param locals_dict: personalized localsdict
         """
         self.ensure_one()
 
-        locals_dict = {
-            'sum': _sum,
-            'min': _min,
-            'max': _max,
-            'len': len,
-            'avg': _avg,
-            'AccountingNone': AccountingNone,
-            'SimpleArray': SimpleArray,
-        }
+        # prepare the localsdict
+        if locals_dict is None:
+            locals_dict = {}
+            locals_dict.update(self.prepare_localsdict())
```

Comment:
I would do this update outside of the `if`, as I see no use case for not having sum/min/max/AccountingNone etc in the locals dict.